### ISSUE TITLE
Optimize for size/network requests

### ISSUE THREAD

**Issue by Ristovski** [ionic-team/stencil#2962]
`@stencil/core@2.6.0`

Maybe related: https://github.com/ionic-team/stencil/issues/683, https://github.com/ionic-team/stencil/issues/2506

There is currently no (documented) way of optimizing `stencil` for size or number of fetch requests done.

For example, generating the standard `ionic-pwa` starter results in 109 javascript files in the `www/build` folder (which ends up being a whopping 6.6MB).

To make matters worse,  `ionicons` get copied ad verbatim to the `www/build/svg` folder, even when they are not used - it would be nice the build (`prod`) step could copy over only the _**used**_ icons, instead of all of them which weigh in at 5.3MB.

I have also noticed that the js output, which is supposed to be fully minified - has newlines and spaces unstripped.
Running `uglify` over all the generated javascript code leads to a 147KB size reduction.

Comment https://github.com/ionic-team/stencil/issues/683#issuecomment-736956003 mentioned usage of `rollup` along with the `dist-custom-elements-bundle` - but I am unsure whether this is useful for full blown PWAs.

TL;DR: Make PWAs smaller by 1) bundling ionic/stencil into bigger modules to minimize number of requests 2) bundle only used `ionicons`.

Ideas/solutions very welcome.

**Comment by cjorasch** [comment]
Some other related issues:

Icons as pure string values or functional components can be very efficient.  This avoids round trips and allows standard js bundling to only include icons that are used.  It also handles efficient versioning and caching which is not easily supported with icons in a directory.  Most of the code in `<ion-icon>` is related to loading and caching the values but in practice the total size of the icons that are used in an app is so small that you end up with a smaller bundle even when the icons are inlined.

https://github.com/ionic-team/ionicons/issues/536

Optimizing for http round trips can be very important:

https://github.com/ionic-team/stencil/issues/2376

**Comment by splitinfinities** [comment]
Hey there, thank you for the patience getting back to you. The new team is getting started and we're working through the backlog now.

I think we are lacking in documentation. Have you used bundles in the stencil.config.ts file? This will help you to chunk the generated files together in order to let you have more granular control over loading. https://github.com/ionic-team/ionic-framework/blob/main/core/stencil.config.ts#L11

There's much more to your comment, so I'd like to open this up with a question - what, from your perspective, about Stencil's compiler and it's performance should be better documented and shared to community members?

**Comment by splitinfinities** [comment]
Also - if you could add any reproduction repo's to help us learn and understand the performance impact you're mentioned, please don't hesitate to leave those here!
